contrib/hamilton/contrib/user/zilto/llm_generate_code/__init__.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import os
+import subprocess
+from typing import Optional
+
+from hamilton.function_modifiers import extract_fields
+
+logger = logging.getLogger(__name__)
+
+from hamilton import contrib
+
+with contrib.catch_import_errors(__name__, __file__, logger):
+    import openai
+
+
+def llm_client(api_key: Optional[str] = None) -> openai.OpenAI:
+    """Create an OpenAI client"""
+    if api_key is None:
+        api_key = os.environ.get("OPENAI_API_KEY")
+
+    return openai.OpenAI(api_key=api_key)
+
+
+def prompt_template_to_generate_code() -> str:
+    return """Write some {code_language} code to solve the user's problem.
+
+Return only python code in Markdown format, e.g.:
+
+```{code_language}
+....
+```
+
+user problem
+{query}
+
+{code_language} code
+"""
+
+
+def prompt_to_generate_code(
+    prompt_template_to_generate_code: str, query: str, code_language: str = "python"
+) -> str:
+    return prompt_template_to_generate_code.format(
+        query=query,
+        code_language=code_language,
+    )
+
+
+def response_generated_code(llm_client: openai.OpenAI, prompt_to_generate_code: str) -> str:
+    response = llm_client.completions.create(
+        model="gpt-3.5-turbo-instruct",
+        prompt=prompt_to_generate_code,
+    )
+    return response.choices[0].text
+
+
+def generated_code(response_generated_code: str) -> str:
```

Comment:
parsed_generated_code ?

Reply:
funnily enough, that's the initial name I had!